### ISSUE TITLE
Livery Categories and Folders - Suggestion

### ISSUE THREAD

**Issue by Moonspire** [TeamOpenIndustry/ImmersiveRailroading#1395]
**Describe the suggestion**  
So I remember discussing this with Cam a long time ago but I was told by Hax that I should translate this to a ticket so here we go.

In IR right now we have liveries and they are great but when you get more than 10 options it can get rather complex. It would be nice to allow model makers the option to create livery folders for organizing livery options.

Along the same lines it would be fantastic to allow for players to select different liveries for each texture file used in a model to allow for models with more complex options to simplify the use for players and reduce redundant textures across multiple liveries. For the purposes of explanation I am calling these "Categories".

**Applications of the suggestion**  
What we have right now when you use the paint brush on a train

```
- Livery 1
- Livery 2
- Livery 3
- ...
```

What I am proposing (select 1 option from each category, categories could have folders to organize options within that category))

```
Category 1
- Option 1
- Option 2
- ...

Category 2
Folder 1
- Option 1
- Option 2
- ...
Folder 2
- Option 1
- Option 2
- ...
```

**Comment by Moonspire** [comment]
Categories and folders would also allow for showing/hiding of their contents in the paint gui

**Comment by FireStorm525** [comment]
Good idea. I think this should be applied to track styles as well because those get flooded really quick
